plugins/out_s3/s3.c

```diff
@@ -2104,10 +2127,29 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
     struct s3_file *upload_file = NULL;
     struct flb_s3 *ctx = out_context;
     struct multipart_upload *m_upload_file = NULL;
+    msgpack_unpacked result;
+    msgpack_object *obj;
+    size_t off = 0;
+    struct flb_time tms;
 
     /* Cleanup old buffers and initialize upload timer */
     flush_init(ctx);
 
+    /* unpack msgpack */
+    msgpack_unpacked_init(&result);
+
+    /* Get the first record timestamp */
+    while (msgpack_unpack_next(&result,
+                               event_chunk->data,
+                               event_chunk->size, &off) == MSGPACK_UNPACK_SUCCESS) {
+        flb_time_pop_from_msgpack(&tms, &result, &obj);
+        if (&tms.tm.tv_sec != 0) {
+            break;
+        }
+    }
+
+    msgpack_unpacked_destroy(&result);
```

Comment:
We only need to get timestamp for first flush per chunk S3_file. So, we can run this code only if there is no existing chunk S3_file for this tag...